### ISSUE TITLE
Assert in when clause triggers but is ignored

### ISSUE THREAD

**Issue by bilderbuchi** [OpenModelica/OpenModelica#9208]
### Description
An assert in a `when` clause does not work correctly -- it triggers, but is subsequently ignored.

### Steps to Reproduce
Run
```
model AssertDoesNotAbort "Assert in when equation is ignored"
Integer i;
equation
  when time >= 0.5 then
    i = 5;
    print("Print " + String(time));
    assert(false, "Assert in half-time clause");
  end when;
end AssertDoesNotAbort;
```
to get this output:
```
The initialization finished successfully without homotopy method.
The following assertion has been violated at time 0.500000
false
Assert in half-time clause
Found event, previous asserts are ignored.
### STATISTICS ###
timer
events
    1 state events
    0 time events
solver: euler
The simulation finished successfully.
Print 0.5
```
* The assert is triggered correctly at the correct time, its message is displayed.
* However, the assert is ignored with "Found event, previous asserts are ignored.", and the simulation finished successfully.

### Expected Behavior
The assert is not ignored when it fires, and reliably aborts the simulation.

Possibly related: #6195, #8200?

### Version and OS
OpenModelica 1.19.0, Windows 10

**Comment by AnHeuermann** [comment]
Related to #8118 / #6195.
The reasoning was that if we search for an event asserts can happen that will be avoided once the event time is known and the event iteration changes whatever caused the assert in the first place.

So if we have something like (a bad example, but whatever)
```Modelica
when x >= 1
  x = pre(x) - 1;
  assert(x <= 1, "x to larger");
end when;
```
we will trigger asserts, but once the event is resolved everything is fine.

But in this specific case we should check, that no assert is thrown at the event point (so after we found the event). If that is the case we need to fail.

**Comment by casella** [comment]
I agree, we should wait until the event has been processed completely, and then check all the asserts and trigger them if necessary.

**Comment by bilderbuchi** [comment]
> I agree, we should wait until the event has been processed completely, and then check all the asserts and trigger them if necessary.

@casella   @AnHeuermann we are still seeing something like this in a model... can this check-asserts-after-event-processing be fixed?

**Comment by casella** [comment]
@bilderbuchi, unfortunately @AnHeuermann has left the developers' group. I'll pass this onto @phannebohm.
